hack/runtime-migrator/cmd/migration.go

```diff
@@ -43,29 +44,26 @@ func NewMigration(migratorConfig config2.Config, converterConfig config.Converte
 	}, nil
 }
 
-func (m Migration) Do(runtimeIDs []string) error {
+func (m Migration) Do(ctx context.Context, runtimeIDs []string) error {
 
-	migratorContext, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	listCtx, cancel := context.WithTimeout(ctx, timeoutK8sOperation)
 	defer cancel()
 
-	shootList, err := m.shootClient.List(migratorContext, v1.ListOptions{})
+	shootList, err := m.shootClient.List(listCtx, v1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
 	results := migration.NewMigratorResults(m.outputWriter.NewResultsDir)
 
 	reportError := func(runtimeID, shootName string, msg string, err error) {
-		var errorMsg string
 
 		if err != nil {
-			errorMsg = fmt.Sprintf("%s: %v", msg, err)
-		} else {
-			errorMsg = fmt.Sprintf(msg)
+			msg = fmt.Sprintf("%s: %v", msg, err)
```

Comment:
It will change the behaviour as there is the following call:
```
reportError(runtimeID, "", "Failed to find shoot", nil)
```

this is not a big deal since the error will be "Failed to find shoot: nil"